src/NServiceBus.Core/Transports/Learning/LearningTransportMessagePump.cs

```diff
@@ -67,32 +67,30 @@ public Task Initialize(PushRuntimeSettings limitations, Func<MessageContext, Tas
             return Task.CompletedTask;
         }
 
-        public Task StartReceive()
+        public Task StartReceive(CancellationToken cancellationToken)
         {
-            cancellationTokenSource = new CancellationTokenSource();
+            stopRequestedCancellationTokenSource = new CancellationTokenSource();
 
-            cancellationToken = cancellationTokenSource.Token;
-
-            messagePumpTask = Task.Run(ProcessMessages, cancellationToken);
+            messagePumpTask = Task.Run(() => ProcessMessages(stopRequestedCancellationTokenSource.Token), stopRequestedCancellationTokenSource.Token);
```

Comment:
Not sure either but since Task.Run takes a token shouldn't we pass it there just to be safe?

Reply:
The more important question is:

What is the observable behavior you want to achieve from the perspective of the one that awaits the `messagePumpTask`. When you pass the token to `Task.Run` the task will switch its state to canceled when the token is canceled and thus awaiting that task will throw an operation canceled exception regardless of the outcome of the `ProcessMessages` method. That might not be the desired effect since for graceful shutdown purposes you might want to just shut down when the token is canceled but not rethrow the actual exception. So it boils down to the design philosophy and on which level you want to force doing the try-catch in case of a graceful shutdown.

Reply:
As per https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.run?view=net-5.0#System_Threading_Tasks_Task_Run_System_Func_System_Threading_Tasks_Task__System_Threading_CancellationToken_ the token arg cancels the task _if it has not yet started_.

Reply:
Nice, I thought this happens always but as this documentation shows it only happens when the task is not yet started. Learned something!